3_harmonize/src/clean_temperature_data.R

```diff
@@ -0,0 +1,44 @@
+#' @title Clean temperature data
+#' 
+#' @description 
+#' Function to clean temperature data, including harmonizing diverse units. 
+#' 
+#' @param wqp_data data frame containing the data downloaded from the WQP, 
+#' where each row represents a data record.
+#' @param char_names_crosswalk data frame containing columns "char_name" and 
+#' "parameter". The column "char_name" contains character strings representing 
+#' known WQP characteristic names associated with each parameter.
+#' @param temp_param_name character string indicating which string in the 
+#' "parameter" column of `char_names_crosswalk` corresponds with temperature
+#' data. 
+#' 
+#' @returns 
+#' Returns a formatted and harmonized data frame containing data downloaded from 
+#' the Water Quality Portal, where each row represents a data record. Temperature
+#' units have been standardized to degrees celsius where possible. 
+#' 
+clean_temperature_data <- function(wqp_data, char_names_crosswalk, temp_param_name){
+  
+  # Grab characteristic names associated with temperature in `char_names_crosswalk`
+  temp_char_names <- char_names_crosswalk %>%
+    filter(parameter == temp_param_name) %>%
+    pull(char_name)
+  
+  # Clean temperature data
+  wqp_data_out <- wqp_data %>%
+    # harmonize units
+    mutate(ResultMeasureValue = if_else(CharacteristicName %in% temp_char_names & 
+                                          !is.na(ResultMeasureValue) & 
+                                          ResultMeasure.MeasureUnitCode == "deg F",
```

Comment:
Do we want a way to have a list of potential unit codes passed in representing Fahrenheit? I imagine there is more than just this one, but we also don't want to have to catch them all in this example.

Reply:
> Do we want a way to have a list of potential unit codes passed in representing Fahrenheit?

Are you referring to USGS Pcodes? Or just different values of `ResultMeasure.MeasureUnitCode` that correspond to degrees Fahrenheit?

Reply:
Was thinking different values of `ResultMeasure.MeasureUnitCode` that correspond to degrees Fahrenheit, e.g. `F`, `degrees F`, etc

Reply:
I've only seen "deg F" so far, so I didn't make it more complex than that. I'd expect we could catch others by changing this line to `grepl("deg F", ResultMeasure.MeasureUnitCode, ignore.case = TRUE)` instead though. Do you know of other examples that wouldn't be captured by that change?

Reply:
I went ahead and added some flexibility to this search for "deg F" values in https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/75/commits/3e457d171edf264c55e1e10c278dbadcc322017c.